Assets/Scenes/Shared Scenes/TextToSpeechHandler.cs

```diff
@@ -160,24 +156,29 @@ private void SpeakNativeCurrentWord(SpeakEventArgs e)
         //
         if(speakingSentence)
         {
-            //
+            // The variable tick will be 0 when the previous tile is done
+            // When that is the case...
             if (tick == 0)
             {
-                //
+                // Progress to the next word tile (tracked by index)
                 index++;
 
-                //
-                tick = words[index].word.word.Split(' ').Length;
+                // The text on this tile
+                WordTile wt = wordTiles[index] as WordTile;
+                string textToRead =  wt.textToDisplay;
+
+                // Calculate the number of ticks this tile will get depending upon how many words are on the tile
+                tick = textToRead.Split(' ').Length;
 
                 //
                 if (index > 0)
                 {
                     //
-                    words[index - 1].Highlight();
+                    wordTiles[index - 1].Highlight(Speaker.ApproximateSpeechLength(textToRead));
```

Comment:
I'm not sure if changes in this line of code actually impact the world or not becuase highlighting does not work for me on my computer anyhow...